### PR TITLE
Add docs on zeroshot image classification prompt templates

### LINKED ISSUE #30951: google/siglip-so400m-patch14-384 inference output mismatch with pipeline output
### System Info

- `transformers` version: 4.41.0
- Platform: Windows-10-10.0.19041-SP0
- Python version: 3.9.13
- Huggingface_hub version: 0.23.0
- Safetensors version: 0.4.3
- Accelerate version: 0.30.1
- Accelerate config:    - compute_environment: LOCAL_MACHINE
        - distributed_type: NO
        - mixed_precision: fp16
        - use_cpu: False
        - debug: False
        - num_processes: 1
        - machine_rank: 0
        - num_machines: 1
        - gpu_ids: all
        - rdzv_backend: static
        - same_network: True
        - main_training_function: main
        - enable_cpu_affinity: False
        - downcast_bf16: no
        - tpu_use_cluster: False
        - tpu_use_sudo: False
        - dynamo_config: {'dynamo_backend': 'INDUCTOR'}
- PyTorch version (GPU?): 2.3.0+cu121 (True)
- Tensorflow version (GPU?): 2.16.1 (False)
- Flax version (CPU?/GPU?/TPU?): not installed (NA)
- Jax version: not installed
- JaxLib version: not installed
- Using GPU in script?: Yes
- Using distributed or parallel set-up in script?: No

### Who can help?

@amyeroberts 

### Information

- [X] The official example scripts
- [ ] My own modified scripts

### Tasks

- [X] An officially supported task in the `examples` folder (such as GLUE/SQuAD, ...)
- [ ] My own task or dataset (give details below)

### Reproduction

Using sample code from https://huggingface.co/google/siglip-so400m-patch14-384:

```
from PIL import Image
import requests
from transformers import AutoProcessor, AutoModel
import torch

model = AutoModel.from_pretrained("google/siglip-so400m-patch14-384")
processor = AutoProcessor.from_pretrained("google/siglip-so400m-patch14-384")

url = "http://images.cocodataset.org/val2017/000000039769.jpg"
image = Image.open(requests.get(url, stream=True).raw)

texts = ["a photo of 2 cats", "a photo of 2 dogs"]
inputs = processor(text=texts, images=image, padding="max_length", return_tensors="pt")

with torch.no_grad():
    outputs = model(**inputs)

logits_per_image = outputs.logits_per_image
probs = torch.sigmoid(logits_per_image) # these are the probabilities
print(f"{probs[0][0]:.1%} that image 0 is '{texts[0]}'")

```

The output mismatches with the pipeline approach:
```
from transformers import pipeline
from PIL import Image
import requests

# load pipe
image_classifier = pipeline(task="zero-shot-image-classification", model="google/siglip-so400m-patch14-384")

# load image
url = 'http://images.cocodataset.org/val2017/000000039769.jpg'
image = Image.open(requests.get(url, stream=True).raw)

# inference
outputs = image_classifier(image, candidate_labels=["2 cats", "a plane", "a remote"])
outputs = [{"score": round(output["score"], 4), "label": output["label"] } for output in outputs]
print(outputs)
```
and the difference is massive.
Pipeline approach seem to give the right results, which also align with inference API.

### Expected behavior

Correct result for the manual approach

### PR DESCRIPTION
**Pull request by aliencaocao**
# What does this PR do?

Adds documentation on prompt templates for Zero Shot Image Classification tasks so that the output results match pipelines and are also correct. Not applying the template can lead to greatly worsened accuracy.

Fixes #30951 

## Before submitting
- [x] This PR fixes a typo or improves the docs (you can dismiss the other checks if that's the case).
- [ ] Did you read the [contributor guideline](https://github.com/huggingface/transformers/blob/main/CONTRIBUTING.md#create-a-pull-request),
      Pull Request section?
- [x] Was this discussed/approved via a Github issue or the [forum](https://discuss.huggingface.co/)? Please add a link
      to it if that's the case.
- [x] Did you make sure to update the documentation with your changes? Here are the
      [documentation guidelines](https://github.com/huggingface/transformers/tree/main/docs), and
      [here are tips on formatting docstrings](https://github.com/huggingface/transformers/tree/main/docs#writing-source-documentation).
- [ ] Did you write any new necessary tests?

## Who can review?
@amyeroberts

### CODE REVIEW

**Comment by aliencaocao** [comment]
i have a PR that adds support for siglip training coming up so would be great if this could be merged asap so the docs don't merge conflict

**Comment by amyeroberts** [comment]
> i have a PR that adds support for siglip training coming up so would be great if this could be merged asap so the docs don't merge conflict

@aliencaocao I appreciate that you want to have PRs merged in as soon as possible (we do too!) but managing conflicts for other PRs isn't something that we will consider, unless it's also for transformers something we want to merge in and linked to.

File: docs/source/en/model_doc/siglip.md [hunk 1]
```diff
@@ -29,6 +29,7 @@ The abstract from the paper is the following:
 - Usage of SigLIP is similar to [CLIP](clip). The main difference is the training loss, which does not require a global view of all the pairwise similarities of images and texts within a batch. One needs to apply the sigmoid activation function to the logits, rather than the softmax.
 - Training is not yet supported. If you want to fine-tune SigLIP or train from scratch, refer to the loss function from [OpenCLIP](https://github.com/mlfoundations/open_clip/blob/73ad04ae7fb93ede1c02dc9040a828634cb1edf1/src/open_clip/loss.py#L307), which leverages various `torch.distributed` utilities.
 - When using the standalone [`SiglipTokenizer`] or [`SiglipProcessor`], make sure to pass `padding="max_length"` as that's how the model was trained.
+- A basic prompt template of "This is a photo of {label}." should be used. This is handled automatically in Pipelines.
```
Comment (amyeroberts) [inline docs/source/en/model_doc/siglip.md:32]: I don't think this is quite right - any prompt could be used. It's just that this prompt will provide the numbers in the example. 

In the pipeline, it's only handled in the zero-shot case, AFAIK and can be overridden, this is just a default setting.

File: docs/source/en/model_doc/siglip.md [hunk 2]
```diff
@@ -81,7 +83,8 @@ If you want to do the pre- and postprocessing yourself, here's how to do that:
 >>> url = "http://images.cocodataset.org/val2017/000000039769.jpg"
 >>> image = Image.open(requests.get(url, stream=True).raw)
 
->>> texts = ["a photo of 2 cats", "a photo of 2 dogs"]
+>>> candidate_labels = ["a photo of 2 cats", "a photo of 2 dogs"]
+>>> candidate_labels = [f'This is a photo of {label}.' for label in candidate_labels]  # follows the training data template else the output will be off
```
Comment (amyeroberts) [inline docs/source/en/model_doc/siglip.md:87]: This is would produce:

`["This is a photo of a photo of 2 cats", "This is a photo of a photo of 2 dogs"]`

which I'm pretty sure we don't want

Comment (amyeroberts) [inline docs/source/en/model_doc/siglip.md:87]: It's not that the output will be off, it's that these will reproduce the outputs from the pipeline

File: docs/source/en/tasks/zero_shot_image_classification.md [hunk 3]
```diff
@@ -118,7 +118,8 @@ Use the processor to prepare the inputs for the model. The processor combines an
 image for the model by resizing and normalizing it, and a tokenizer that takes care of the text inputs.
 
 ```py
->>> candidate_labels = ["tree", "car", "bike", "cat"]
+>>> candidate_labels = ["tree", "car", "bike", "cat"
+>>> candidate_labels = [f'This is a photo of {label}.' for label in candidate_labels]  # follows the training data template else the output will be off
```
Comment (amyeroberts) [inline docs/source/en/tasks/zero_shot_image_classification.md:122]: Same here

**Review by amyeroberts — COMMENTED** [review]
Thanks for contributing to improving the docs! 

I'm important to distinguish between what the user must feed to the model, and a useful example. The "This is a photo of {}" prompt just helps replicate the values from the pipeline, but it's not a requirement and can in fact be changed in the pipeline too.

File: docs/source/en/model_doc/siglip.md (hunk 1 above)
Comment (aliencaocao) [reply docs/source/en/model_doc/siglip.md:32]: Ok, I think I will rephrase to: "To get the same results as the pipeline, a basic prompt template of "This is a photo of {label}." should be used."
I was thinking more of fine-tuning cases and its weird that you deviate from the pretrained prompt format when finetuning.

File: docs/source/en/model_doc/siglip.md (hunk 2 above)
Comment (aliencaocao) [reply docs/source/en/model_doc/siglip.md:87]: Yes my bad, will fix

Comment (aliencaocao) [reply docs/source/en/model_doc/siglip.md:87]: Will edit the comment

**Comment by aliencaocao** [comment]
@amyeroberts heads up i've made the changes already

File: docs/source/en/tasks/zero_shot_image_classification.md [hunk 4]
```diff
@@ -119,6 +119,7 @@ image for the model by resizing and normalizing it, and a tokenizer that takes c
 
 ```py
 >>> candidate_labels = ["tree", "car", "bike", "cat"]
+>>> candidate_labels = [f'This is a photo of {label}.' for label in candidate_labels]  # follows the pipeline prompt template to get same results
```
Comment (amyeroberts) [inline docs/source/en/tasks/zero_shot_image_classification.md:122]: nit

```suggestion
# follows the pipeline prompt template to get same results
>>> candidate_labels = [f'This is a photo of {label}.' for label in candidate_labels]
```

File: docs/source/en/model_doc/siglip.md [hunk 5]
```diff
@@ -81,7 +83,8 @@ If you want to do the pre- and postprocessing yourself, here's how to do that:
 >>> url = "http://images.cocodataset.org/val2017/000000039769.jpg"
 >>> image = Image.open(requests.get(url, stream=True).raw)
 
->>> texts = ["a photo of 2 cats", "a photo of 2 dogs"]
+>>> candidate_labels = ["2 cats", "2 dogs"]
+>>> candidate_labels = [f'This is a photo of {label}.' for label in candidate_labels]  # follows the pipeline prompt template to get same results
```
Comment (amyeroberts) [inline docs/source/en/model_doc/siglip.md:87]: nit 

```suggestion
# follows the pipeline prompt template to get same results
>>> candidate_labels = [f'This is a photo of {label}.' for label in candidate_labels]
```

**Review by amyeroberts — APPROVED** [review]
Thanks! 

Just two tiny nits

**Comment by aliencaocao** [comment]
Merged your suggestions

**Comment by amyeroberts** [comment]
@aliencaocao Can you try rebasing to include the latest changes upstream? This should resolve some of the issues on the CI runs

**Comment by aliencaocao** [comment]
@amyeroberts hi somehow the workflows arent getting triggered?

**Comment by amyeroberts** [comment]
@aliencaocao We've had a few issues with tests triggering on circle CI today. I'm not 100% sure of the cause, but other PRs which have blocked workflows are now working. Could you try triggering a new run? `git commit --allow-empty -m "Trigger CI"`

**Comment by aliencaocao** [comment]
@amyeroberts its ready now

**Comment by amyeroberts** [comment]
Thanks for updating the docs and your patience with the CI!
